benches/cub/device/memcpy/basic.cu

```diff
@@ -0,0 +1,358 @@
+#include <nvbench/detail/throw.cuh>
+#include <nvbench/nvbench.cuh>
+
+#include <thrust/detail/raw_pointer_cast.h>
+#include <thrust/device_vector.h>
+#include <thrust/execution_policy.h>
+#include <thrust/host_vector.h>
+#include <thrust/sequence.h>
+
+#include <cub/device/device_memcpy.cuh>
+#include <cub/iterator/transform_input_iterator.cuh>
+
+#include <cstdint>
+#include <limits>
+#include <random>
+#include <stdexcept>
+
+enum class BufferOrder
+{
+  // Buffers are randomly shuffled within memory
+  RANDOM,
+
+  // Buffer N+1 resides next to buffer N
+  CONSECUTIVE
+};
+
+/**
+ * @brief Function object class template that takes an offset and returns an
+ * iterator at the given offset relative to a fixed base iterator.
+ *
+ * @tparam IteratorT The random-access iterator type to be returned
+ */
+template <typename IteratorT>
+struct OffsetToPtrOp
+{
+  template <typename T>
+  __host__ __device__ __forceinline__ IteratorT operator()(T offset) const
+  {
+    return base_it + offset;
+  }
+  IteratorT base_it;
+};
+
+/**
+ * @brief Host-side random data generation
+ */
+template <typename T>
+void GenerateRandomData(
```

Comment:
Would [existing ](https://github.com/allisonvacanti/thrust_benchmark/blob/0494020cdb7d1e4e1962a8d2583de044b3e1f4a5/benches/cub/device/radix_sort/sort_pairs_bench.cuh#L44) random generator suit you? I guess you could copy generated data from GPU to CPU.

Reply:
I've tried to replace it with tbm's generator but hit some rough edges that required post-processing and normalizing the generated numbers, ultimately introducing similar code overhead. Hence, decided to keep the current solution.